data/transactions/logic/assembler.go

```diff
@@ -1086,79 +1111,176 @@ func asmDefault(ops *OpStream, spec *OpSpec, args []string) error {
 				}
 				ops.pending.WriteByte(val)
 			}
+		case immInt8:
+			val, err := int8Imm(args[i], imm.Name)
+			if err != nil {
+				return ops.errorf("%s %w", spec.Name, err)
+			}
+			ops.pending.WriteByte(val)
 		default:
 			return ops.errorf("unable to assemble immKind %d", imm.kind)
 		}
 	}
 	return nil
 }
 
-// Interprets the arg at index argIndex as byte-long immediate
-func getByteImm(args []string, argIndex int) (byte, bool) {
+// getImm interprets the arg at index argIndex as an immediate
+func getImm(args []string, argIndex int) (int, bool) {
 	if len(args) <= argIndex {
 		return 0, false
 	}
-	n, err := strconv.ParseUint(args[argIndex], 0, 8)
+	// We want to parse anything from -128 up to 255. So allow 9 bits.
+	// Normal assembly checkign willl catch signed as byte, vice versa
```

Comment:
```suggestion
	// Normal assembly checking will catch signed as byte, vice versa
```

Reply:
thanks, I'll commit the change as I remove the `stacky` function.